### DISCUSSION TITLE
Minimal example doesn't work under Windows

### DISCUSSION THREAD

**Discussion by leontoeides** [GitoxideLabs/prodash#16] [Q&A]
Hey there! Prodash looks awesome. Pretty excited to start using it but there is some confusion on how to get the renderer working.

Using the minimal example from here: https://docs.rs/prodash/21.0.0/prodash/render/tui/index.html

I updated the `render_fut` clause to:

```rust
// Configure the gui, provide it with a handle to the ever-changing tree
let render_fut = prodash::render::tui::render(
    std::io::stdout(),
    Arc::downgrade(&root),
    prodash::render::tui::Options {
        title: "minimal example".into(),
        ..prodash::render::tui::Options::default()
    }
)?;
```

When I run the example the screen is blank. Any advice? Thank you!

**Comment by Byron** [comment]
I recommend trying to run the example like this:

```sh
cargo run --example dashboard --features="render-tui render-tui-crossterm render-line render-line-crossterm signal-hook render-line-autoconfigure progress-tree"
```

It comes up with the TUI renderer by default and if it works, you can check it's code to see how it's brought up. Otherwise it's something fundamental related to the terminal you are using, it might not work there.

You can also add these arguments to the line above to try the `line` renderer like so: `-- -R line`, which should work in more terminals. Please note that it's brought up differently from the `tui` renderer.

I hope that helps.

**Comment by leontoeides** [comment]
I should've excluded "Windows" in the title, I just wanted to make it clear it could be a "Windows thing" - or it might not be.

Running from the repo fork `--example` works perfectly!

The "problem" for me is that the `example` directory is too intense - too difficult & time consuming to pull apart to get started quickly and involves dependencies I'd prefer to not bring in. I hope it doesn't seem like I'm asking you to do my homework for me... I'm sure other potential users would appreciate more minimal examples.

The minimal example that I linked to doesn't seem to work for me and it appears outdated. Of course, it's likely a user error. Is it possible to confirm that the minimal example still works correctly before I knock myself out on it?

**Reply by Byron** [reply]
If the example is from the docs than at least it compiles, but it's not guaranteed that it's still correct. As unfortunately I don't have the time right now to provide support in such detail, I hope you will manage to find your way and eventually contribute a PR with the example you would have loved to see.

Thanks for your help and understanding.

**Reply by leontoeides** [reply]
No worries! I feel that way all the time :-) Thank you for your responsiveness despite that. If I can make a better minimal example, I will send it along.
